File: VotingSystem/src/voting/ElectoralBody.java
package voting;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class ElectoralBody {
    public static List<Candidate> candidateList = new ArrayList<Candidate>();
    private List<Voter> voterList = new ArrayList<>();
    private static List<Long> voterIdList = new ArrayList<>();
    private static List<Integer>candidateIdList = new ArrayList<>();

    private int numberOfCandidates;
    private int numberOfVoters;
    private String position;

    public ElectoralBody(String position) {
        this.position = position;
    }

    public ElectoralBody() {

    }

    public Candidate registerCandidate(String firstName, String lastName, String position) {
       int candidateId = numberOfCandidates++;
       candidateIdList.add(candidateId);
       String name = firstName + " " + lastName;
       Candidate candidate = new Candidate(name, position, candidateId);
       candidateList.add(candidate);
       return candidate;
    }

    public Voter registerVoter(String firstName, String lastName, String password){
        numberOfVoters++;
        long voterId = generateVoterId();
        voterIdList.add(voterId);
        String name = firstName + " " + lastName;
        Voter voter = new Voter(name, password, voterId);
        voterList.add(voter);
        return voter;
    }

    public static List<Integer> getCandidateIdList(){
        return candidateIdList;
    }

    public static List<Long> getVoterIdList(){
        return voterIdList;
    }

    public int getCandidateId() {
        return numberOfCandidates;
    }


    public int getNumberOfRegisteredVoters() {
        return numberOfVoters;
    }

    public long generateVoterId(){
        Random rand = new Random();
        long voterId = Math.abs(rand.nextLong() % 2_147_483_647);
        return voterId;
    }

    public long getVoterId(){
        return voterIdList.get(voterIdList.size()-1);
    }

    public String getPosition(){
        return position;
    }

    public String findCandidateNameByPosition(String position){
        for (Candidate candidate : candidateList) {
            if (candidate.getPosition().equals(position)) {
                return candidate.getName();
            }
        }
        throw new IllegalArgumentException("The position does not exist");
    }

    public String getCandidateName(int candidateId){
        for (Candidate candidate : candidateList) {
            if (candidate.getId() == candidateId) {
                return candidate.getName();
            }
        }
        throw new IllegalArgumentException("The candidate does not exist");
    }


}

File: VotingServiceJs/ElectoralBody.js
const Candidate = require(`./Candidate.js`);
const Voter = require(`./Voter.js`);

class ElectoralBody{

    #candidateIdList = [];
    #candidates = [];
        #voters = [];
        #voterIdList = [];
    #numberOfCandidates;

    constructor(){
        this.#numberOfCandidates = 0;
    }

    registerCandidate(firstName, lastName, position){
        let candidate_id = this.#numberOfCandidates+=1;
        this.#candidateIdList.push(candidate_id);
        let name = "firstName"+ " " + "lastName";
        let newCandidate = new Candidate(name, position, candidate_id);
        this.#candidates.push(newCandidate);
        return newCandidate;
    }

    registerVoter(firstName, lastName, password){
        let voterId = this.generateVoterId();
        this.#voterIdList.push(voterId);
        let name = "firstName"+" " + "lastName";
        let newVoter = new Voter(name, password, voterId, this);
        this.#voters.push(newVoter);
        return newVoter;
    }

    generateVoterId(){
        const minCeiled = Math.ceil(1_200_000);
        const maxFloored = Math.floor(1_999_999);
        return Math.floor(Math.random() * (maxFloored - minCeiled + 1) + minCeiled);
    }

    get candidateIdList(){
        return this.#candidateIdList;
    }

    get voterIdList(){
        return this.#voterIdList;
    }



}

module.exports = ElectoralBody;